src/main/webapp/WEB-INF/view/admin.jsp

```diff
@@ -13,26 +13,12 @@
   See the License for the specific language governing permissions and
   limitations under the License.
 --%>
-<%@ page import="codeu.model.data.Conversation" %>
-<%@ page import="codeu.model.data.Message" %>
-<%@ page import="codeu.model.data.User" %>
-<%@ page import="codeu.model.store.basic.ConversationStore" %>
-<%@ page import="codeu.model.store.basic.MessageStore" %>
-<%@ page import="codeu.model.store.basic.UserStore" %>
+<%@ page import="java.util.HashMap" %>
+<%@ page import="java.util.Map" %>
 
 <!DOCTYPE html>
 
-<%
-/** Retrieve an instance of each Datastore */
-ConversationStore conversationStore = ConversationStore.getInstance();
-UserStore userStore = UserStore.getInstance();
-MessageStore messageStore = MessageStore.getInstance();
-
-/** Retrieve sizes of each Datastore */
-Integer convSize = conversationStore.countTotalConversations();
-Integer userSize = userStore.countTotalUsers();
-Integer messageSize = messageStore.countTotalMessages();
-%>
+<% Map<String, String> adminStatsMap = (HashMap<String, String>) request.getAttribute("adminStatsMap");%>
```

Comment:
Use Map interface here too. Method clients should not need to know the concrete implementation class of the Map

Reply:
Done!